include/seqan3/alignment/pairwise/alignment_result.hpp

```diff
@@ -210,36 +210,68 @@ class alignment_result
         return data.score;
     }
 
-    /*!\brief Returns the back coordinate of the alignment.
+    /*!\brief Returns the end position of the first sequence of the alignment.
+     * \return The calculated alignment end of sequence 1 (inclusive).
+     *
+     * \note This function is only available if the end position of the first sequence was requested via the
+     * alignment configuration (see seqan3::align_cfg::result).
+     */
+    constexpr size_t sequence1_end_position() const noexcept
+    {
+        static_assert(!std::is_same_v<size_t, std::nullopt_t *>,
```

Comment:
Sorry I found one more thing. This statement here makes no sense because it is always true.
see https://github.com/Irallia/seqan3/pull/5 (I thought this was more tricky first because I did not use `auto` but needed to lazily get the type and then I already changed everything so with PR it is not wasted)